### PR TITLE
feat(backend): endpoint for GET txn email message by id

### PR DESCRIPTION
**Pull request by stanleynguyen**
## Problem

[Ticket](https://www.notion.so/opengov/Implement-get-transactional-message-by-id-endpoint-99343591ccef4665a2dbfb841790a0ab)

## Solution

Implement the endpoint

## Deployment Checklist

_Any tasks that need to be done on the environment before releasing of this PR (e.g. Setting environment variables, running migrations, etc). If there's no task to be done, put "N/A"_

- N/A

### CODE REVIEW

File: backend/src/email/routes/email-transactional.routes.ts [hunk 1]
```diff
@@ -224,7 +229,7 @@ export const InitEmailTransactionalRoute = (
    *          type: string
    *
    */
-  router.use(
+  router.post(
```
Comment (tan-zx) [inline backend/src/email/routes/email-transactional.routes.ts:232]: ohh good catch

curious why was it `router.use()` previously hmm

File: backend/src/email/interfaces/email.interface.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ export interface EmailDuplicateCampaignDetails
  *          - status
  *        properties:
  *          id:
- *            type: number
+ *            type: string
```
Comment (tan-zx) [inline backend/src/email/interfaces/email.interface.ts:44]: just thinking out loud — I read that in API design, we should "be liberal in what you accept, be conservative in what you send". do you think it's a good idea for us to be able to accept both and convert accordingly?

File: backend/src/email/routes/email-transactional.routes.ts [hunk 3]
```diff
@@ -32,6 +32,11 @@ export const InitEmailTransactionalRoute = (
       attachments: Joi.array().items(Joi.object().keys().required()),
     }),
   }
+  const getByIdValidator = {
+    [Segments.PARAMS]: Joi.object({
+      emailId: Joi.number().required(),
```
Comment (tan-zx) [inline backend/src/email/routes/email-transactional.routes.ts:37]: same as above about being liberal in what we accept 🤔

File: backend/src/email/models/email-message-transactional.ts [hunk 4]
```diff
@@ -37,7 +37,7 @@ export class EmailMessageTransactional extends Model<EmailMessageTransactional>
     primaryKey: true,
     allowNull: false,
   })
-  id: number
+  id: string
```
Comment (tan-zx) [inline backend/src/email/models/email-message-transactional.ts:40]: this one can just change without any db migration? 😯

**Review by tan-zx — COMMENTED** [review]
A few things:
- Wonder whether we should accept `emailId` as both string and number (see comments in code)
- I'll be pushing a small PR that lightly edits the Swagger docs (typos and stuff); pls help me review that!

Just to document our discussion on whether it's OK to expose a monotonically incrementing `emailId` to the user.
- Basically, exposing this information allows the user to infer how many transactional email messages are being sent out. Not sure whether this is sensitive.
- In more commercial settings, this could reveal sensitive financial information (but not very relevant to us). In theory, any "competing" product can use this to deduce how much usage we have.
- We are exposing the total number of messages sent via each channel on the landing page (though that statistic does not include transactional emails). 

I think we can probably discuss this at a team level? 🤔

File: backend/src/email/middlewares/email-transactional.middleware.ts [hunk 5]
```diff
@@ -39,6 +40,23 @@ export const InitEmailTransactionalMiddleware = (
     attachments?: { data: Buffer; name: string; size: number }[]
   }
 
+  function convertMessageModalToResponse(message: EmailMessageTransactional) {
```
Comment (tan-zx) [inline backend/src/email/middlewares/email-transactional.middleware.ts:43]: typo, should be `MessageModel` I think

File: backend/src/email/routes/email-transactional.routes.ts [hunk 6]
```diff
@@ -369,7 +357,7 @@ export const InitEmailTransactionalRoute = (
    *                example:
    *                  {status: 429, message: Too many requests. Please try again later.}
    *         "500":
-   *           description: Internal Server Error (includes error such as custom domain passed email validation but is incorrect)
+   *           description: Internal Server Error
```
[the remark this answers is not in the document]
Comment (stanleynguyen) [reply backend/src/email/routes/email-transactional.routes.ts:360]: Oh wait u're removing that

**Comment by stanleynguyen** [comment]
> A few things:
> 
> * Wonder whether we should accept `emailId` as both string and number (see comments in code)
> * I'll be pushing a small PR that lightly edits the Swagger docs (typos and stuff); pls help me review that!
> 
> Just to document our discussion on whether it's OK to expose a monotonically incrementing `emailId` to the user.
> 
> * Basically, exposing this information allows the user to infer how many transactional email messages are being sent out. Not sure whether this is sensitive.
> * In more commercial settings, this could reveal sensitive financial information (but not very relevant to us). In theory, any "competing" product can use this to deduce how much usage we have.
> * We are exposing the total number of messages sent via each channel on the landing page (though that statistic does not include transactional emails).
> 
> I think we can probably discuss this at a team level? 🤔

Practically, we have no reason to worry about our IDs being incremental. This concern is **over-paranoia**

File: backend/src/email/interfaces/email.interface.ts (hunk 2 above)
Comment (stanleynguyen) [reply backend/src/email/interfaces/email.interface.ts:44]: Not sure why u're commenting on this irrelevant line

File: backend/src/email/routes/email-transactional.routes.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (tan-zx) [reply backend/src/email/routes/email-transactional.routes.ts:360]: yeah I'm removing it from the other routes (the original one for `/transactional/email/send` is still there though)

I just searched and this and the 413 stuff I deleted are in `email-settings.routes.ts` too, should probably remove them hmm

File: backend/src/email/interfaces/email.interface.ts (hunk 2 above)
Comment (stanleynguyen) [reply backend/src/email/interfaces/email.interface.ts:44]: Moving it to where it's relevant

File: backend/src/email/models/email-message-transactional.ts (hunk 4 above)
Comment (stanleynguyen) [reply backend/src/email/models/email-message-transactional.ts:40]: The previous data type declaration, number, is wrong. Number in JS can't fit SQL BIGINT

File: backend/src/email/routes/email-transactional.routes.ts (hunk 3 above)
Comment (stanleynguyen) [reply backend/src/email/routes/email-transactional.routes.ts:37]: Joi is already automatically converting numerical strings to number

File: backend/src/email/models/email-message-transactional.ts (hunk 4 above)
Comment (tan-zx) [reply backend/src/email/models/email-message-transactional.ts:40]: Ohh I see, makes sense!

**Review by tan-zx — APPROVED** [review]

**Pull request merged**